Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m

```diff
@@ -111,6 +212,90 @@ + (instancetype)providerWithAuth:(FIRAuth *)auth {
 }
 
 #pragma mark - Internal Methods
+/** @fn isVerifyAppURL:
+    @brief Parses a URL into all available query items.
+    @param URL The url to be checked against the authType string.
+    @return Whether or not the URL matches authType.
+ */
+- (BOOL)isVerifyAppURL:(NSURL *)URL {
+  NSArray *strings = [_auth.app.options.clientID componentsSeparatedByString:@"."];
+  NSString *reverseClientID =
+      [[strings reverseObjectEnumerator].allObjects componentsJoinedByString:@"."];
+  if (![[URL scheme] isEqualToString:reverseClientID]) {
+    return NO;
+  }
+  if (![[URL host] isEqualToString:kFirebaseAuthHost]) {
+    return NO;
+  }
```

Comment:
Please check path, too. And also port, user, and password to be cautious.

Alternatively, you can reconstruct the incoming URL without query and fragment by using NSURLComponent, and then compare to the expected URL directly.

Reply:
Done.